### PR TITLE
Update predicate

### LINKED ISSUE #16: Ability to provide custom comparison function to see if interested in update
Often a controller is not interested in an update (e.g. just status change), and if it's expensive to resync we really don't want to be processing it.

Either Process should give us the old object and the new object, or we should have the option of giving a custom comparison function to ctrl so we only see interesting updates.

### PR DESCRIPTION
**Pull request by ash2k**
Fixes #16.

### CODE REVIEW

**Comment by amckague** [comment]
Testing?

File: types.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ type ZapNameField func(name string) zap.Field
 type Descriptor struct {
 	// Group Version Kind of objects a controller can process.
 	Gvk schema.GroupVersionKind
+	// IsInterestingUpdate is an optional predicate that is consulted before enqueuing an object on update event.
+	IsInterestingUpdate func(oldObj, newObj runtime.Object) bool
```
Comment (jokeyrhyme) [inline types.go:26]: "Interesting" is fun, but I wonder if maybe something like `ShouldUpdate()` is more descriptive?

**Review by benbarclay — APPROVED** [review]

**Comment by ash2k** [comment]
I've been thinking about an alternative (more flexible) approach where we let the controller constructor attach whatever event handler it wants to use. Approach in this PR does not feel right, it is not generic enough...

File: types.go (hunk 1 above)
Comment (ychen-atlassian) [reply types.go:26]: Was going to comment the exact same thing before I saw your comment on another line 👍

**Pull request closed without merging**